### PR TITLE
Add missing data for the element function

### PR DESCRIPTION
**Pull request by stof**
updated matching https://caniuse.com/#feat=css-element-function

### CODE REVIEW

**Comment by stof** [comment]
note that there is an inconsistency between Caniuse and BCD regarding the support in Firefox:

- Caniuse says there is full (prefixed) support in Firefox 4+ and support in some properties only in Firefox 2+
- BCD says that support in 4+ is limited to some properties only

Which one is right ?

**Comment by ddbeck** [comment]
I fell down a bit of rabbit hole here trying to find an answer to the Firefox question (caniuse someitmes links to relevant issues, but not in this case, so I considered it suspect). Here's what I learned:

I don't think that `-moz-element` was ever released before Firefox 4 (based on [this Hacks post](https://hacks.mozilla.org/2010/08/mozelement/)). Seems like the pre-version 4 support that caniuse reports is spurious.

It does appear that the limitation to `background`/`background-image` persisted beyond Firefox 4, but I'm not sure when that support expanded ended on a per-property basis. But [bug 700926](https://bugzilla.mozilla.org/show_bug.cgi?id=700926) provides some clues.

[This commit](https://hg.mozilla.org/mozilla-central/rev/c4af6c845f68) makes it appear that `border-image(-source)` got support in Firefox 29. And [this comment](https://bugzilla.mozilla.org/show_bug.cgi?id=700926#c65) implies that `-moz-element` should work anywhere `<image>` works as of Firefox 57/60 (i.e., when Stylo shipped on desktop and Android, respectively).

My poorly educated guess here is that a more correct set of notes would say something like:

* From Firefox 29 to Firefox 57 (aside: the version would be 60 for the Android entry), `-moz-element` was only supported with the `background`, `background-image`, `border-image`, and `border-image-source` properties.
* From Firefox 4 to 29, `-moz-element` was only supported on the `background` and `background-image` properties.

**Comment by stof** [comment]
@ddbeck should I update the Firefox data based on that educated guess ?

**Comment by ddbeck** [comment]
Yes, that'd be great! Sorry, I should have made that clearer.

**Comment by stof** [comment]
@ddbeck updated.

I was not sure whether I should add a `version_removed` on older notes (so that ranges go after each other). And there is no way to preview the result, as `npm run render` is outdated (#736)

File: css/types/image.json [hunk 1]
```diff
@@ -1639,24 +1639,46 @@
...
               },
+              "firefox": [
+                {
+                  "prefix": "-moz-",
+                  "version_added": "57"
+                },
+                {
+                  "prefix": "-moz-",
+                  "version_added": "29",
+                  "notes": "<code>element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/border-image'><code>border-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/border-image-source'><code>border-image-source</code></a>."
```
Comment (ddbeck) [inline css/types/image.json]:
```suggestion
                  "notes": "<code>-moz-element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/border-image'><code>border-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/border-image-source'><code>border-image-source</code></a>."
```

File: css/types/image.json [hunk 2]
```diff
@@ -1639,24 +1639,46 @@
...
+                },
+                {
+                  "prefix": "-moz-",
+                  "version_added": "4",
+                  "notes": "<code>element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>."
```
Comment (ddbeck) [inline css/types/image.json]:
```suggestion
                  "notes": "<code>-moz-element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>."
```

File: css/types/image.json [hunk 3]
```diff
@@ -1639,24 +1639,46 @@
...
+              ],
+              "firefox_android": [
+                {
+                  "prefix": "-moz-",
+                  "version_added": "60"
+                },
+                {
+                  "prefix": "-moz-",
+                  "version_added": "29",
+                  "notes": "<code>element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/border-image'><code>border-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/border-image-source'><code>border-image-source</code></a>."
```
Comment (ddbeck) [inline css/types/image.json]:
```suggestion
                  "notes": "<code>-moz-element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/border-image'><code>border-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/border-image-source'><code>border-image-source</code></a>."
```

File: css/types/image.json [hunk 4]
```diff
@@ -1639,24 +1639,46 @@
...
+                },
+                {
+                  "prefix": "-moz-",
+                  "version_added": "4",
+                  "notes": "<code>element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>."
```
Comment (ddbeck) [inline css/types/image.json]:
```suggestion
                  "notes": "<code>-moz-element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>."
```

File: css/types/image.json [hunk 5]
```diff
@@ -1639,24 +1639,46 @@
...
-                "notes": "<code>element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>."
+                "version_added": false
               },
+              "firefox": [
+                {
+                  "prefix": "-moz-",
+                  "version_added": "57"
+                },
+                {
+                  "prefix": "-moz-",
```
Comment (ddbeck) [inline css/types/image.json]: This is a great update. Seeing this spelled out now, I think we probably need to add this:

```suggestion
                  "prefix": "-moz-",
                  "partial_implementation": true,
```

Sorry for not realizing this sooner!

File: css/types/image.json [hunk 6]
```diff
@@ -1639,24 +1639,46 @@
...
+                  "version_added": "29",
+                  "notes": "<code>element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/border-image'><code>border-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/border-image-source'><code>border-image-source</code></a>."
+                },
+                {
+                  "prefix": "-moz-",
```
Comment (ddbeck) [inline css/types/image.json]:
```suggestion
                  "prefix": "-moz-",
                  "partial_implementation": true,
```

File: css/types/image.json [hunk 7]
```diff
@@ -1639,24 +1639,46 @@
...
+                  "notes": "<code>element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>."
+                }
+              ],
+              "firefox_android": [
+                {
+                  "prefix": "-moz-",
+                  "version_added": "60"
+                },
+                {
+                  "prefix": "-moz-",
```
Comment (ddbeck) [inline css/types/image.json]:
```suggestion
                  "prefix": "-moz-",
                  "partial_implementation": true,
```

File: css/types/image.json [hunk 8]
```diff
@@ -1639,24 +1639,46 @@
...
+                  "version_added": "29",
+                  "notes": "<code>element()</code> is limited to <a href='https://developer.mozilla.org/docs/Web/CSS/background-image'><code>background-image</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/background'><code>background</code></a>, <a href='https://developer.mozilla.org/docs/Web/CSS/border-image'><code>border-image</code></a> and <a href='https://developer.mozilla.org/docs/Web/CSS/border-image-source'><code>border-image-source</code></a>."
+                },
+                {
+                  "prefix": "-moz-",
```
Comment (ddbeck) [inline css/types/image.json]:
```suggestion
                  "prefix": "-moz-",
                  "partial_implementation": true,
```

File: css/types/image.json (hunk 6 above)
Comment (stof) [reply css/types/image.json]: this is not a partial implementation for the final version though, is it ?

**Comment by stof** [comment]
@ddbeck now updated with your suggestions and with `version_removed`

**Pull request merged**